### PR TITLE
Fixed Evaluation

### PR DESCRIPTION
**Pull request by abrutti**
In train() (in utils/train.py) evaluation on the test set is run on the best model (not the last one) according to the accuracy on dev set

load_audio() in classe SpeechDataset modified to always use the cached audio for test and dev sets. Otherwise results change when using the loader multiple time, even if using the same model. shuffle set to False for test and dev loaders.

### CODE REVIEW

File: utils/train.py [hunk 1]
```diff
@@ -110,12 +111,12 @@ def train(config):
...
         dev_set,
         batch_size=min(len(dev_set), 16),
-        shuffle=True,
+        shuffle=False,
         collate_fn=dev_set.collate_fn)
     test_loader = data.DataLoader(
         test_set,
-        batch_size=min(len(test_set), 16),
-        shuffle=True,
+        batch_size=len(test_set),
```
Comment (daemon) [inline utils/train.py:118]: Can we stick to `min(len(test_set), 16)`? The entire test set isn't guaranteed to fit in the VRAM, for different choices of `test_pct`.

File: utils/train.py [hunk 2]
```diff
@@ -151,15 +152,15 @@ def train(config):
...
-                loss_numeric = loss.item()
                 accs.append(print_eval("dev", scores, labels, loss))
             avg_acc = np.mean(accs)
             print("final dev accuracy: {}".format(avg_acc))
             if avg_acc > max_acc:
                 print("saving best model...")
                 max_acc = avg_acc
                 model.save(config["output_file"])
-    evaluate(config, model, test_loader)
+                best_model=copy_deepcopy(model)
```
Comment (daemon) [inline utils/train.py]: Where is `copy_deepcopy` defined?

**Review by daemon — CHANGES_REQUESTED** [review]

File: utils/train.py (hunk 1 above)
Comment (abrutti) [reply utils/train.py:118]: Of course it does not affect the performance

File: utils/train.py (hunk 2 above)
Comment (abrutti) [reply utils/train.py]: copy.deepcopy(model), sorry.
```suggestion
                best_model=copy.deepcopy(model)
```

**Comment by daemon** [comment]
Thanks for your contribution.
